lib/mix/lib/mix/tasks/format.ex

```diff
@@ -651,11 +651,23 @@ defmodule Mix.Tasks.Format do
     mix format failed due to --check-formatted.
     The following files are not formatted:
 
-    #{to_bullet_list(not_formatted)}
+    #{to_diffs(not_formatted)}
     """)
   end
 
-  defp to_bullet_list(files) do
-    Enum.map_join(files, "\n", &"  * #{&1 |> to_string() |> Path.relative_to_cwd()}")
+  defp to_diffs(files) do
+    Enum.map_join(files, "\n", fn
+      {:stdin, unfomatted, fomatted} ->
+        IO.iodata_to_binary([IO.ANSI.reset(), Mix.TextDiff.format(unfomatted, fomatted)])
```

Comment:
I'm probably overthinking this, but just wondering: maybe we don't need to cast it as a binary here and we can build an IO list?
- removing calls to `IO.iodata_to_binary/1`
- replacing `map_join/3` by `map_intersperse/3`